resolver/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+//! TODO
+
+use commonware_cryptography::Array;
+use std::future::Future;
+
+pub mod p2p;
+
+/// The interface that gets notified when data is available.
+pub trait Consumer: Clone + Send + 'static {
+    /// Type used to uniquely identify data.
+    type Key: Array;
+
+    /// Type of data to retrieve.
+    type Value;
+
+    /// Type used to indicate why data is not available.
+    type FailureCode;
```

Comment:
I think we can just call this `Failure` (it may or may not be a Code and that shouldn't be our issue).